qiskit_experiments/library/characterization/cr_hamiltonian_analysis.py

```diff
@@ -0,0 +1,294 @@
+
+        bounds = {
+            "px0": user_bounds["px0"] or (-np.inf, np.inf),
+            "py0": user_bounds["py0"] or (-np.inf, np.inf),
+            "pz0": user_bounds["pz0"] or (-np.inf, np.inf),
+            "px1": user_bounds["px1"] or (-np.inf, np.inf),
+            "py1": user_bounds["py1"] or (-np.inf, np.inf),
+            "pz1": user_bounds["pz1"] or (-np.inf, np.inf),
+            "b": user_bounds["b"] or (-1, 1),
+        }
+
+        guesses = defaultdict(list)
+
+        for control in (0, 1):
+            # start from Z oscillation
+            x_data = self._data(series_name=f"x|c={control}")
+            y_data = self._data(series_name=f"y|c={control}")
+            z_data = self._data(series_name=f"z|c={control}")
+
+            omega_xyz = []
+            for data in (x_data, y_data, z_data):
+                fft_freq = curve.guess.frequency(data.x, data.y)
+                # oscillation amplitude might be almost zero, then exclude from average
+                if fft_freq > 0:
+                    omega_xyz.append(fft_freq)
+            if omega_xyz:
+                omega = 2 * np.pi * np.average(omega_xyz)
+            else:
+                omega = 0.0
+
+            zmin, zmax = np.percentile(z_data.y, [10, 90])
+            theta = np.arccos(np.sqrt((zmax - zmin) / 2))
+
+            # The FFT might be up to 1/2 bin off
+            df = 1 / ((z_data.x[1] - z_data.x[0]) * len(z_data.x))
+            for omega_shifted in [omega, omega - df / 2, omega + df / 2]:
+                for phi in np.linspace(-np.pi, np.pi, 9):
+                    px = omega_shifted * np.cos(theta) * np.cos(phi)
+                    py = omega_shifted * np.cos(theta) * np.sin(phi)
+                    pz = omega_shifted * np.sin(theta)
+                    guesses[control].append(
+                        {
+                            f"px{control}": user_p0[f"px{control}"] or px,
+                            f"py{control}": user_p0[f"py{control}"] or py,
+                            f"pz{control}": user_p0[f"pz{control}"] or pz,
+                        }
+                    )
+
+        fit_options = []
+        for p0s, p1s in zip(guesses[0], guesses[1]):
+            fit_option = {
+                "p0": {**p0s, **p1s, "b": user_p0["b"] or 1e-9},
+                "bounds": bounds,
+            }
+            fit_option.update(extra_options)
+            fit_options.append(fit_option)
+
+        return fit_options
+
+    def _evaluate_quality(self, fit_data: curve.FitData) -> Union[str, None]:
+        """Algorithmic criteria for whether the fit is good or bad.
+
+        A good fit has:
+            - All fit parameter uncertainties are smaller than fit values.
+        """
+        criteria = []
+        for control in (0, 1):
+            for axis in ("x", "y", "z"):
+                fit_par = fit_data.fitval(f"p{axis}{control}")
+                criteria.append(fit_par.value > fit_par.stderr)
+
+        if all(criteria):
```

Comment:
I don't think this works for deciding the quality because sometimes the parameters are small. For example for a well tuned drive, ZY and IY should both be zero. Looking at chi squared might be the best that can be done.

Also, sometimes CR Hamiltonian tomography is used to measure stray coupling and cross-talk. In these cases, you are hoping that all the terms are small. Maybe that will need to be a separate experiment any way? I don't think the way control channel is looked up in the experiment (`pulse.control_channels`) will work for disconnected qubit pairs any way?

Reply:
Your right. There was the same suggestion from @eggerdj https://github.com/Qiskit/qiskit-experiments/pull/385#discussion_r708810852

I noticed multi object function fit used in qiskit tends to report higher chi-sq value. See the attached figure in the PR header, the chi-sq value is around 16 though the curve agrees well with the data points. What is the criteria of chi-sq value in this case? For experiments with a single curve seems like we conventionally use chi_sq < 3.